### PR TITLE
Window mode: intermediate switch to desktop only in awesome

### PR DESCRIPTION
**Pull request by lbonn**
As reported in #537, switching to the target's desktop before the final
window destination can have unintended consequences, so it's best to avoid
it if possible.

However, it is necessary inside awesome wm (#393). This commit
explicitely checks the wm's name to choose the window switching method.

### CODE REVIEW

File: source/dialogs/window.c [hunk 1]
```diff
@@ -587,42 +628,33 @@ static ModeMode window_mode_result ( Mode *sw, int mretv, G_GNUC_UNUSED char **i
...
 
-                                                          wmdesktop,
-                                                          XCB_CURRENT_TIME );
+                wm_name = window_get_text_prop ( wm_win, xcb->ewmh._NET_WM_NAME );
+
+                if ( wm_name == NULL ) {
+                    wm_name = window_get_text_prop ( wm_win, XCB_ATOM_WM_NAME );
+                }
+
+                if ( wm_name ) {
```
Comment (sardemff7) [inline source/dialogs/window.c]: We use GLib, you can safely go with `g_strcmp0(wm_name, "awesome") == 0`

**Comment by DaveDavenport** [comment]
We currently have two different detection of WM. one for I3 (by looking if a property is there) and one for awesomeWM.

I think it might be a good idea to write a WM detection function that runs on startup. I will look at that this weekend.

**Comment by lbonn** [comment]
Sure, this should be cleaner.

This PR uses a method similar to what is done by `wmctrl -m`. I tested it under i3 and awesome.

**Comment by lbonn** [comment]
The wm detection that was pushed in master does not seem to work reliably under
awesome. Apparently i3 adds its name to the root but awesome does not.

Under awesome:

```
$ xwininfo -tree -root

xwininfo: Window id: 0x105 (the root window) (has no name)

  Root window id: 0x105 (the root window) (has no name)
  Parent window id: 0x0 (none)
     20 children:
     ...

     0x600007 "Awesome no input window": ("awesome" "awesome")  1x1+-1+-1
     0x600006 "Awesome systray window": ("awesome" "awesome")  1x1+-1+-1  +-1+-1
     0x600005 "awesome": ("awesome" "awesome")  1x1+-1+-1  +-1+-1
     0x600004 "Awesome WM_Sn selection owner window": ("awesome" "awesome")
     ...
```

Under i3:

```
$ xwininfo -tree -root

xwininfo: Window id: 0x105 (the root window) "i3"

  Root window id: 0x105 (the root window) "i3"
  Parent window id: 0x0 (none)
     28 children:
     ...
     0x60006a "[i3 con] workspace __i3_scratch": ("i3-frame" "i3-frame")  10x10+-15+-15  +-15+-15
     0x600068 "[i3 con] content __i3": ("i3-frame" "i3-frame")  10x10+-15+-15  +-15+-15
     0x600066 "[i3 con] pseudo-output __i3": ("i3-frame" "i3-frame")  10x10+-15+-15  +-15+-15
     0x600064 (has no name): ("i3-frame" "i3-frame")  10x10+-15+-15  +-15+-15
     0x600063 "i3": ()  1x1+-1+-1  +-1+-1
     ...
```

I believe that the call to `xcb_ewmh_get_supporting_wm_check_xxx` is necessary
to get a window with the wm's name if it exists.

**Comment by DaveDavenport** [comment]
ok. will add.. 
I tried a couple wms I had installed for those it worked..

**Comment by DaveDavenport** [comment]
Added, can you check.

**Comment by DaveDavenport** [comment]
ok. That seems to work.

**Pull request closed without merging**